scripts/publish_video/helper.js

```diff
@@ -307,11 +342,35 @@ async function delegateHP(account, hp) {
   );
 }
 
+async function shouldSkip(video) {
+  if (video.status === 'publish_manual') {
+    if (video.fromMobile === true ) {
+      try {
+        let doWeHavePostingAuthority = await hasPostingAuthority(video.owner);
+        if (doWeHavePostingAuthority === false) {
+          return true;
+        }
+        video.status = 'published';
+        await video.save();
```

Comment:
To me shouldSkip implies it's a checking function to read a certain condition not modify a value. Might be misued and accidentally change something in the DB (i.e publish a video). I'd suggest putting the checking in a separate function and the actual writing separately.